hydra-node/src/Hydra/Chain/Direct/State.hs

```diff
@@ -983,43 +1028,70 @@ genCollectComTx = do
   let spendableUTxO = getKnownUTxO stInitialized
   pure (cctx, committedUTxO, stInitialized, unsafeCollect cctx headId (ctxHeadParameters ctx) utxoToCollect spendableUTxO)
 
+genDecrementTx :: Int -> Gen (ChainContext, OpenState, Tx)
+genDecrementTx numParties = do
+  ctx <- genHydraContextFor numParties
+  (u0, stOpen@OpenState{headId}) <- genStOpen ctx
+  cctx <- pickChainContext ctx
+  let version = 0
+  snapshot <- do
+    number <- getPositive <$> arbitrary
+    let (utxo, toDecommit) = splitUTxO u0
+    pure Snapshot{headId, number, confirmed = [], utxo, utxoToDecommit = Just toDecommit, version}
+  let signatures = aggregate $ fmap (`sign` snapshot) (ctxHydraSigningKeys ctx)
+  let openUTxO = getKnownUTxO stOpen
+  pure (cctx, stOpen, unsafeDecrement cctx headId (ctxHeadParameters ctx) openUTxO snapshot signatures)
+
+splitUTxO :: UTxO -> (UTxO, UTxO)
+splitUTxO utxo =
+  let pairs = UTxO.pairs utxo
+   in case uncons $ List.filter ((> 0) . selectLovelace . txOutValue . snd) pairs of
+        Nothing -> (mempty, mempty)
+        Just (toDecommit, rest) -> (UTxO.fromPairs rest, UTxO.singleton toDecommit)
+
 genCloseTx :: Int -> Gen (ChainContext, OpenState, Tx, ConfirmedSnapshot Tx)
 genCloseTx numParties = do
   ctx <- genHydraContextFor numParties
   (u0, stOpen@OpenState{headId}) <- genStOpen ctx
-  snapshot <- genConfirmedSnapshot headId 0 u0 (ctxHydraSigningKeys ctx)
+  let (confirmedUtxo, utxoToDecommit) = splitUTxO u0
+  let version = 0
+  snapshot <- genConfirmedSnapshot headId 1 version confirmedUtxo (Just utxoToDecommit) (ctxHydraSigningKeys ctx)
```

Comment:
This is hard, because the generator is defined generically over `tx`. Maybe we should have a generic one to use in `arbitrary` and a specific one for these kinds of tests? Anyways.. not now.